### ISSUE TITLE
Make search instructions accessible

### ISSUE THREAD

**Issue by klemay** [hypothesis/product-backlog#823]
When using our search function, the user sees instructions for narrowing their search:

<img width="668" alt="screen shot 2018-10-05 at 10 46 37 am" src="https://user-images.githubusercontent.com/12087705/46542276-03d3f800-c88c-11e8-9456-ab30478aa09a.png">

However, these instructions disappear after the search parameter is typed. For instance, when searching by url, the instructions disappear as soon as the user types `url:`

![url-search](https://user-images.githubusercontent.com/12087705/46542316-21a15d00-c88c-11e8-9841-94388b1ee3ec.gif)

For a user with cognitive difficulties (or really, any user who isn't familiar with how our search works!) this makes search difficult to use.

Also, from the perspective of someone using the screenreader, I'm not sure whether the contents of that dropdown with instructions are available. 

We would likely need input from @dwolfe on alternative ways to present these instructions, and we could assess those from an a11y standpoint.

**Comment by dwolfe** [comment]
Yep, as @hmstepanek and I were talking about this issue, I thought that we might be finding the limits of the instructions dropdown. Also, @robertknight mentioned to me that the code for this feature is a little fiddly in general, and could use some refactoring.

Recording an idea here: maybe the dropdown could be contextual, such that it first shows something like _Narrow your search by user:, tag:, url:, or group:_, and when one of those keywords is typed, additional instructions for that modifier appear. Conditional on the existence of an accessible way to do the dropdown.

**Comment by ritobruno** [comment]
Hello, I'm new here and I noticed that this issue has been inactive for some time.
I agree that it's hard to notice the instruction while typing, so my question is, are they necessary? The user can search and get results without using the suggestions.

**Comment by klemay** [comment]
Hi @ritobruno! Yes, this issue has been inactive because this part of our code will need to be refactored before we can make meaningful improvements. 

The user can indeed continue to type and perform a search even though the instructions disappear - this issue is asking for a usability/accessibility improvement.

**Comment by ritobruno** [comment]
Thanks @klemay. 
I'll wait for the code refactor and meanwhile, I will think and maybe design some usability improvements for this component. Is it the right time for this, or should I wait for green light of someone?

**Comment by klemay** [comment]
@ritobruno we're a small team currently focused on other areas of the code, so unfortunately I wouldn't be able to guarantee that we'd be able to do a code review for something like this - however we welcome suggestions for usability improvements in the form of user stories / feature requests! those could be shared at any time as a comment on this issue.

**Comment by ritobruno** [comment]
Okay @klemay, as a new user of Hypothesis and designer I'm searching for something to start my collaboration with you.
Any advice?

**Comment by klemay** [comment]
@ritobruno since we're such a small org we're not really set up to manage a collaboration like this. if you're looking for ways to contribute to open source projects more generally, I'd suggest searching github for tags like "help wanted" to find open source projects that can review and include outside contributions. As we continue to grow we may be able to do something like this in the future - we can ping you here if/when that happens.
